site/zenodo_rdm/moderation/handlers.py

```diff
@@ -75,106 +58,136 @@ def rules(self):
             return current_app.config[self._rules]
         return self._rules or []
 
-    def run(self, identity, draft=None, record=None, uow=None):
+    @property
+    def should_apply_actions(self):
+        """Return if moderation actions should be applied."""
+        return current_app.config.get("MODERATION_APPLY_ACTIONS", False)
+
+    @cached_property
+    def exempt_users(self):
+        """Return users exempt from moderation."""
+        return set(current_app.config.get("MODERATION_EXEMPT_USERS", []))
+
+    def run(self, identity, draft=None, record=None, user=None, uow=None):
         """Calculate the moderation score for a given record or draft."""
         try:
+            if user.id in self.exempt_users:
+                current_app.logger.info(
+                    "User is exempt from moderation", extra={"user_id": user.id}
+                )
+                return
+
             score = 0
             for rule in self.rules:
                 score += rule(identity, draft=draft, record=record)
 
-            # TODO: Move to caller handler method (i.e. publish, create, update, etc.)
-            user_id = None
-            if identity == system_identity:
-                # Use the owner of the record as the user for moderation actions
-                if isinstance(record, records_service.record_cls):
-                    user_id = record.parent.access.owned_by.owner_id
-            else:
-                user_id = identity.id
-            if user_id is None:
-                current_app.logger.error(
-                    "No user found for moderation action", stack_info=True
-                )
-                return
-
-            user = UserAggregate.get_record(user_id)
             action_ctx = {
-                "user_id": user_id,
+                "user_id": user.id,
                 "record_pid": record.pid.pid_value,
                 "score": score,
             }
-            apply_actions = current_app.config.get("MODERATION_APPLY_ACTIONS", False)
+            current_app.logger.info(
+                f"Moderation score for {user.id} calculated: {score}", extra=action_ctx
```

Comment:
Actually, all the `info` calls should be `debug`... The issue I have is that `extra`s are not logged right when running the local dev server, which we should look into, since they can be very useful.